File: ArtworkPostServices/src/main/java/com/artauction/artworkpostservices/Controllers/CategoryController.java
package com.artauction.artworkpostservices.Controllers;

import com.artauction.artworkpostservices.Services.Interfaces.CategoryServices;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/Category")
@RequiredArgsConstructor
public class CategoryController {
    // Attributes
    private final CategoryServices categoryServices;

    // Get all categories API
    @GetMapping("/GetAllCategories")
    public ResponseEntity<?> getAllCategories() {
        return ResponseEntity.ok(categoryServices.getAllCategories());
    }
}

File: Project/Backend/src/ArtAuction.API/Controllers/CategoryController.cs
using ArtAuction.Application.Interfaces.Services;
using Microsoft.AspNetCore.Mvc;

namespace ArtAuction.API.Controllers;

[ApiController]
[Route("[controller]")]
public class CategoryController : ControllerBase
{
    // Attributes
    private readonly ICategoryServices _categoryServices;
    
    // Constructor
    public CategoryController(ICategoryServices categoryServices)
    {
        this._categoryServices = categoryServices;
    }

    // Get all categories API
    [HttpGet("GetAllCategories")]
    public async Task<IActionResult> GetAllCategories()
    {
        // Return stats with categories
        return Ok(await _categoryServices.GetAllCategories());
    }
}
